src/uct/tcp/tcp_ep.c

```diff
@@ -237,10 +239,19 @@ void uct_tcp_ep_move_ctx_cap(uct_tcp_ep_t *from_ep, uct_tcp_ep_t *to_ep,
     uct_tcp_ep_add_ctx_cap(to_ep, ctx_cap);
 }
 
+int uct_tcp_ep_failed_remove_filter(const ucs_callbackq_elem_t *elem,
+                                    void *arg)
+{
+    uct_tcp_ep_t *ep = (uct_tcp_ep_t*)arg;
+
+    ucs_assert(ep->flags & UCT_TCP_EP_FLAG_FAILED);
```

Comment:
what if we have some other unrelated elem on the cbq

Reply:
but this filter would be checked agaist all existing elements in cbq, of other transports as well
you can assert this only after checking elem->cb == uct_tcp_ep_failed_progress

Reply:
`arg` passed as a `ucs_callbackq_remove_if()` argument
`elem->arg` couldn't be `ep`, but `arg` is always `ep`

Reply:
ok, right